unified-runtime/source/adapters/level_zero/command_buffer.cpp

```diff
@@ -502,7 +502,9 @@ void ur_exp_command_buffer_handle_t_::cleanupCommandBufferResources() {
   // Release fences allocated to command-buffer
   for (auto &ZeFencePair : ZeFencesMap) {
     auto &ZeFence = ZeFencePair.second;
-    ZE_CALL_NOCHECK(zeFenceDestroy, (ZeFence));
+    if (checkL0LoaderTeardown()) {
```

Comment:
Can move move this outside the `for` loop so we only need to check it once for every iteration, or should it be checked per-iteration?

Reply:
So, because this is a race condition, it is possible that in the middle of the teardown the loader is stable for the first fence destroy, but then is unstable for the next call. 

If we moved it outside, "most" of the time, it should be safe, but there is no guarantee. 

Making this check per destroy call is the only guaranteed method to ensure that we can never call destroy while the loader is unstable.

Reply:
Even with the current implementation, there is a potential for a race (although it's less likely) - the loader might become unstable just after a call to checkL0LoaderTerardown(), right?. I don't know if there is any 100% reliable way of dealing with that.

Reply:
Agreed, this just makes it more likely to address all the known teardown situations with use after free.

We may have a way to fix this in 2.0 of L0, but not today.